adafruit_requests.py

```diff
@@ -529,6 +529,13 @@ def _get_socket(
         self._socket_free[sock] = False
         return sock
 
+    @staticmethod
+    def _header_suppplied(header, supplied_headers):
+        for supplied_header in supplied_headers:
+            if supplied_header.lower() == header.lower():
+                return True
+        return False
```

Comment:
This seems kind of laborious to me, and `.lower()`s each header key multiple times. You could put the `.lower()` version of all the header keys in a `set()`, and then just check set membership.

Reply:
That's a great idea (used to making simple methods in CPython). fixed in https://github.com/adafruit/Adafruit_CircuitPython_Requests/pull/149/commits/5fa5aa218bbaecdd7b81c1e33a0894de635a8eff